### PR TITLE
Retain the attributes when converting delta to/from html

### PR DESCRIPTION
**Pull request by wuzhengu**
When converting delta to/from html, the attributes (eg. the image width, height) are missing

### CODE REVIEW

File: lib/src/packages/quill_markdown/delta_to_markdown.dart [hunk 1]
```diff
@@ -309,17 +306,21 @@ abstract class _NodeVisitor<T> {
...
+      return visitor.visitBlock(node, context);
+    }
+    if (node is Line) {
+      return visitor.visitLine(node, context);
+    }
+    if (node is QuillText) {
+      return visitor.visitText(node, context);
+    }
+    if (node is Embed) {
+      return visitor.visitEmbed(node, context);
```
Comment (ellet0) [inline lib/src/packages/quill_markdown/delta_to_markdown.dart:323]: I think switch statement would be more prefered

**Comment by ellet0** [comment]
Thank you!

File: lib/src/packages/quill_markdown/delta_to_markdown.dart (hunk 1 above)
Comment (wuzhengu) [reply lib/src/packages/quill_markdown/delta_to_markdown.dart:323]: That requires environment sdk ^3.0.0

Comment (ellet0) [reply lib/src/packages/quill_markdown/delta_to_markdown.dart:323]: > That requires environment sdk ^3.0.0

We already requiring SDK 3.0.0

**Comment by ellet0** [comment]
> _ No description provided. _

Can you update the description on details of what this fixes and improves so I can merge this? just to keep things organized in case someone in the future take a look at this

File: lib/src/packages/quill_markdown/delta_to_markdown.dart (hunk 1 above)
Comment (wuzhengu) [reply lib/src/packages/quill_markdown/delta_to_markdown.dart:323]: Where ?

File: lib/src/models/documents/document.dart [hunk 2]
```diff
@@ -458,28 +445,26 @@ class Document {
...
-
-    final mdDocument = md.Document(encodeHtml: false);
+    var rules = [
+      html2md.Rule('image', filters: ['img'], replacement: (content, node) {
+        node.asElement()?.attributes.remove('class');
+        //Later we can convert this to delta along with the attributes by GoodInlineHtmlSyntax
+        return node.outerHTML;
+      }),
+    ];
+
```
Comment (ellet0) [inline lib/src/models/documents/document.dart:455]: Can I know why you are removing the class attributes from the  in HTML??

File: lib/src/packages/quill_markdown/delta_to_markdown.dart (hunk 1 above)
Comment (wuzhengu) [reply lib/src/packages/quill_markdown/delta_to_markdown.dart:323]: https://github.com/singerdmx/flutter-quill/pull/1601#issuecomment-1857744008

Given this, I feel that the environment sdk is finally decided by project pubspec.yaml.

**Comment by ellet0** [comment]
Let me know when you are ready to merge

File: lib/src/models/documents/document.dart (hunk 2 above)
Comment (wuzhengu) [reply lib/src/models/documents/document.dart:455]: It's just no use in markdown

**Comment by wuzhengu** [comment]
> Let me know when you are ready to merge

Already

**Comment by ellet0** [comment]
Did you test it in the quill image sample?

image

**Comment by ellet0** [comment]
Please take your time fixing the bug we just caught, and then we will start on merging the new changes

**Pull request merged**